src/main/java/backend/resource/Model.java

```diff
@@ -99,7 +103,7 @@ public UpdateSignature getUpdateSignature() {
     }
 
     @Override
-    public List<TurboIssue> getIssues() {
+    public synchronized List<TurboIssue> getIssues() {
```

Comment:
But the ultimate wall is that after a new `Model` is created from update data but hasn't been put into the `MultiModel` yet, methods outside that try to modify the issues has no way to access the updated `Model`, meanwhile the updated `Model` has no knowledge of the outside changes and is just gonna squash the changes into the `MultiModel`. This is like a scaled up version of the classic load -> increment -> store mutex problem.